org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/symbols/LSPSymbolInFileDialog.java

```diff
@@ -63,48 +73,98 @@ protected Control createDialogArea(Composite parent) {
 		getShell().setText(NLS.bind(Messages.symbolsInFile, documentFile == null ? null : documentFile.getName()));
 
 		final var filteredTree = new FilteredTree(parent, SWT.BORDER, new PatternFilter(), true, false);
-		TreeViewer viewer = filteredTree.getViewer();
-		viewer.setData(LSSymbolsContentProvider.VIEWER_PROPERTY_IS_QUICK_OUTLINE, Boolean.TRUE);
+		treeViewer = filteredTree.getViewer();
+		treeViewer.setData(LSSymbolsContentProvider.VIEWER_PROPERTY_IS_QUICK_OUTLINE, Boolean.TRUE);
 
-		final var contentService = new NavigatorContentService(CNFOutlinePage.ID, viewer);
+		final var contentService = new NavigatorContentService(CNFOutlinePage.ID, treeViewer);
 		filteredTree.addDisposeListener(ev -> contentService.dispose());
 		final var contentProvider = contentService.createCommonContentProvider();
-		viewer.setContentProvider(contentProvider);
+		treeViewer.setContentProvider(contentProvider);
 
 		final var sorter = new CommonViewerSorter();
 		sorter.setContentService(contentService);
-		viewer.setComparator(sorter);
-
-		viewer.setLabelProvider(new NavigatorDecoratingLabelProvider(contentService.createCommonLabelProvider()));
-		viewer.setUseHashlookup(true);
-
-		final var isFirstSelectionEvent = new AtomicBoolean(true);
-		viewer.addSelectionChangedListener(event -> {
-			// ignore the first selection event which is fired when the current editor
-			// selection is pre-selected in the tree view when the outline popup is opens
-			if (isFirstSelectionEvent.get()) {
-				isFirstSelectionEvent.set(false);
-				return;
+		treeViewer.setComparator(sorter);
+
+		treeViewer.setLabelProvider(new NavigatorDecoratingLabelProvider(contentService.createCommonLabelProvider()));
+		treeViewer.setUseHashlookup(true);
+
+		final Tree tree = treeViewer.getTree();
+
+		// listen to ESC key
+		tree.addKeyListener(new KeyListener() {
+			@Override
+			public void keyPressed(KeyEvent e)  {
+				if (e.character == 0x1B) { // ESC
+					close();
+				}
 			}
+			@Override
+			public void keyReleased(KeyEvent e) {
+				// do nothing
+			}
+		});
 
-			final var selection = (IStructuredSelection) event.getSelection();
-			if (selection.isEmpty()) {
-				return;
+		// listen to double-click or enter key
+		tree.addSelectionListener(new SelectionListener() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				// do nothing
 			}
+			@Override
+			public void widgetDefaultSelected(SelectionEvent e) {
+				gotoSelectedElement();
+			}
+		});
+
+		// listen to left mouse button's single click
+		tree.addMouseListener(new MouseAdapter() {
+			@Override
+			public void mouseUp(MouseEvent e) {
+
+				if (tree.getSelectionCount() < 1)
```

Comment:
could we combine both conditions in one if and use {} even if there is only one statement ?